pkg/chains/formats/slsa/v1/pipelinerun/pipelinerun.go

```diff
@@ -112,14 +113,39 @@ func buildConfig(ctx context.Context, pro *objects.PipelineRunObjectV1Beta1) Bui
 		for _, tr := range taskRuns {
 			// Ignore Tasks that did not execute during the PipelineRun.
 			if tr.Status.CompletionTime == nil {
-				logger.Infof("taskrun status not complete for task %s", tr.Name)
+				logger.Warnf("taskrun status not complete for task %s", tr.Name)
 				continue
 			}
 
 			steps := []attest.StepAttestation{}
-			for i, stepState := range tr.Status.Steps {
-				step := tr.Status.TaskSpec.Steps[i]
-				steps = append(steps, attest.Step(&step, &stepState))
+			// tr.Status.TaskSpec.Steps and tr.Status.Steps should be sime size
+			if len(tr.Status.TaskSpec.Steps) != len(tr.Status.Steps) {
+				logger.Errorf("Mismatch in number of steps for task run %s. TaskSpec steps: %d, Status steps: %d",
+					tr.Name, len(tr.Status.TaskSpec.Steps), len(tr.Status.Steps))
+				continue // Skip this task run entirely
```

Comment:
Shouldn't we mark the `TaskRun` as "failed to sign" or something ? (so that the controller doesn't pick it up later on).

Reply:
I think we dont have a way to do this today, i reference a discussion here about that https://github.com/tektoncd/chains/issues/979